WooCommerce/Classes/ViewRelated/Orders/ManualTrackingViewController.swift

```diff
@@ -0,0 +1,496 @@
+        cell.accessoryType = .none
+    }
+
+    private func configureDateShipped(cell: TitleAndEditableValueTableViewCell) {
+        cell.title.text = NSLocalizedString("Date shipped", comment: "Add / Edit shipping provider. Title of cell date shipped")
+
+        cell.value.text = viewModel.shipmentDate.toString(dateStyle: .medium, timeStyle: .none)
+
+        cell.value.isEnabled = false
+        cell.accessoryType = .none
+    }
+
+    func configureSecondaryAction(cell: BasicTableViewCell) {
+        cell.selectionStyle = .default
+        cell.textLabel?.textAlignment = .center
+        cell.textLabel?.textColor = StyleManager.destructiveActionColor
+        cell.textLabel?.text = viewModel.secondaryActionTitle
+    }
+
+    func configurePicker(cell: DatePickerTableViewCell) {
+        guard datePickerVisible else {
+            return
+        }
+
+        cell.onDateSelected = { [weak self] date in
+            self?.viewModel.shipmentDate = date
+            self?.reload()
+        }
+    }
+}
+
+
+// MARK: - UITableViewDelegate comformance
+//
+extension ManualTrackingViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return CGFloat.leastNonzeroMagnitude
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        let lastSection = sections.count - 1
+        if section == lastSection {
+            return UITableView.automaticDimension
+        }
+        return CGFloat.leastNonzeroMagnitude
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let row = rowAtIndexPath(indexPath)
+
+        if row == .deleteTracking {
+            return Constants.deleteRowHeight
+        }
+
+        guard row == .datePicker else {
+            return Constants.rowHeight
+        }
+
+        guard datePickerVisible else {
+            return CGFloat.leastNonzeroMagnitude
+        }
+
+        return Constants.pickerRowHeight
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        executeAction(for: indexPath)
+    }
+}
+
+
+// MARK: - Actions associated to taps in cells
+//
+private extension ManualTrackingViewController {
+    func executeAction(for indexPath: IndexPath) {
+        let row = rowAtIndexPath(indexPath)
+        if row == .deleteTracking {
+            deleteTracking()
+            return
+        }
+
+        if row == .dateShipped && viewModel.isAdding {
+            displayDatePicker(at: indexPath)
+            return
+        }
+
+        if row == .shippingProvider && viewModel.isAdding {
+            showAllShipmentProviders()
+        }
+    }
+
+    func displayDatePicker(at indexPath: IndexPath) {
+        datePickerVisible = true
+
+        reload()
```

Comment:
What do you think about this [type of presentation](https://medium.com/@tharanit99/how-to-implement-a-inline-date-picker-in-ios-with-swift-4-9f8274460dbc) of the date picker, instead of reloading the table to make it appear? 

(The other issue with reloading the table data is that the tracking number disappears after filling it in and then tapping the date picker.)

Reply:
I tried something similar (reloading just the last row) but it did crash for me. 🤔 most likely because I wasn't doing the reload within a `beginUpdates` and `endUpdates` block. I'll give it another try. FWIW the tracking number does not disappear anymore 😉

Reply:
>the tracking number does not disappear anymore

Hooray 😀